core/serialization_v6.c

```diff
@@ -1049,11 +1050,11 @@ uint32_t chunk_v6_read_shape(Stream *s,
 
                 nameStr = (char *)malloc(nameLen + 1); // +1 for null terminator
                 if (nameStr == NULL) {
-                    totalSizeRead += sizeRead + (uint32_t)sizeof(uint32_t);
-                    continue;
```

Comment:
a continue here wouldn't update the cursor pointer - but if malloc fails you might have other problems, anyway ;)